src/main/java/com/mineinabyss/mobzy/ecs/goals/mobzy/hostile/ThrowItemsGoal.kt

```diff
@@ -95,16 +88,14 @@ class ThrowItemsGoal(
         }
     }
 
-    //TODO try not to rely on NMS at all here
     /** Throws the mob's defined item at the [target]*/
     fun throwItem(target: LivingEntity) {
         val world = mob.location.world ?: return
         val location = mob.eyeLocation
         val (x, y, z) = location
 
-        //TODO some way to create different types of projectiles
-        val projectile = DamagingThrownItem(item, damage, nmsEntity.world, nmsEntity)
-        projectile.setPosition(x, y + yOffset, z)
+        location.y += yOffset
+        val projectile = location.spawnEntity(template.nmsType) ?: return
```

Comment:
I'm still a little unhappy we don't have a consistent way to spawn our mobs. I think it might be worth putting a spawn function on the template itself and always prefer to use that. Could also do `Location.spawn(template)` but in looty we already have an instantiateItemStack function in the template, so I think it makes sense to have a spawn function here as well.